storm-core/src/clj/backtype/storm/config.clj

```diff
@@ -112,6 +124,10 @@
     (FileUtils/forceMkdir (File. ret))
     ret ))
 
+(defn read-supervisor-storm-conf-given-path
+  [conf stormconf-path]
+  (merge conf (clojurify-structure (Utils/fromCompressedJsonConf (FileUtils/readFileToByteArray (File. stormconf-path))))))
```

Comment:
Used it
(defn read-supervisor-storm-conf
  [conf storm-id]
  (let [stormroot (supervisor-stormdist-root conf storm-id)
        conf-path (supervisor-stormconf-path stormroot)
        topology-path (supervisor-stormcode-path stormroot)]
    (read-supervisor-storm-conf-given-path conf conf-path)))